gaphor/storage/storage.py

```diff
@@ -414,3 +418,21 @@ def upgrade_element_to_2_1_0(elem):
             elem.references["comment"] = refids
             del elem.references["ownedComment"]
     return elem
+
+
+def flatten(items):
+    """Yield items from a nested iterable."""
+    for item in items:
+        if isinstance(item, Iterable):
+            yield from flatten(item)
+        else:
+            yield item
+
+
+def upgrade_canvas_item_to_2_1_0(item: gaphas.Item) -> Optional[gaphas.Item]:
+    if item.type == "PartitionItem" and item.canvasitems:
+        item.canvasitems = list(flatten(item.canvasitems))
+        item.canvasitems = [
+            item for _ in item.canvasitems if item.type != "PartitionItem"
+        ]
+    return item
```

Comment:
Is this logic sound? This may result in
`item.canvasitems = [item]`, which causes a circular reference. I suppose what we want is filter out the nested PartitionItems?

Reply:
This is no longer needed because gaphor can now still load old models.